src/components/Admin/Onboard/ProductAddModal.vue

```diff
@@ -159,43 +163,64 @@
         }
         this.create_product_modal_view = true
       },
-      add_product: function () {
-        this.productAddedToast()
-        this.create_product_modal_view = true
-        // ajax call to verify and queue storing
-        this.current_category.products.push({
+      add_ready_product: function () {
+        shop.categoryProductCreate({
           image: this.new_product.image,
-          title: this.new_product.title,
           asset_id: this.new_product.asset_id,
+          store_id: this.current_category.store_id,
+          price_cents: this.new_product.price_cents * 100,
+          category_id: this.current_category.category_id,
           description: this.new_product.description,
-          price_cents: this.new_product.price_cents * 100
+          title: this.new_product.title,
+          display_price: this.new_product.price_cents
+        }).then(response => {
+          this.productAddedToast()
```

Comment:
I think this promise and error handling should probably go in the base instead of returning the promise. That way its ready to use when calling it and the return can be a flag that says whether to go on or not (just a thought)